WEB/BE/utils/totp/index.js

```diff
@@ -0,0 +1,52 @@
+const crypto = require('crypto');
+const base32 = require('hi-base32');
+
+const totp = {
+  makeSecretKey(length = 20) {
+    const secretKey = crypto.randomBytes(length);
+    return base32.encode(secretKey).replace(/=/g, '');
+  },
+
+  makeURL({ secretKey, email }) {
+    return `otpauth://totp/${process.env.SECRETKEYLABEL}?secret=${secretKey}&issuer=${email}`;
+  },
+
+  verifyDigits(key, digits, date = new Date()) {
+    key = base32.decode(key);
+    const sixDigits = makeSixDigits(key, date);
+    return digits === sixDigits;
+  },
+};
+
+const makeSixDigits = (key, date) => {
+  const timeStmap = makeTimeStamp(date);
+  const buffer = makeBuffer(timeStmap);
+  const hash = crypto.createHmac('sha1', key).update(buffer).digest('hex');
+  const DBC = selectDBC(hash);
+  let sixDigits = (parseInt(DBC, 16) % 1000000).toString();
+  sixDigits = sixDigits.length !== 6 ? `0${sixDigits}` : sixDigits;
+  return sixDigits;
+};
+
+const makeTimeStamp = (date, window = 0) => {
+  return Math.floor(new Date(date) / 30000) + window;
+};
+
+const makeBuffer = (timeStmap) => {
+  const buffer = Buffer.alloc(8);
+  for (let i = 0; i < 8; i++) {
+    buffer[7 - i] = timeStmap & 0xff;
+    timeStmap >>= 8;
+  }
+  return buffer;
+};
```

Comment:
`writeUInt32BE`를 사용할 수도 있겠네요!
```suggestion
const makeBuffer = (timeStmap) => {
  const buffer = Buffer.alloc(8);
  buffer.writeUInt32BE(timeStamp, 4); // 앞의 4바이트를 skip
  return buffer;
};
```